### PR TITLE
[988] Add 'IntelliJ Setup' section to README

### LINKED ISSUE #988: Fixed: IntelliJ `DeltaSqlBaseBaseVisitor` type not found
## Problem
A common error I have seen when developing Delta Lake using IntelliJ is the following (below). These errors prevent IntelliJ from running unit tests (though, running unit tests via `build/sbt core/test` still works fine).

```
Error:(46, 28) object DeltaSqlBaseParser is not a member of package io.delta.sql.parser
import io.delta.sql.parser.DeltaSqlBaseParser._
Error:(143, 34) not found: type DeltaSqlBaseBaseVisitor
class DeltaSqlAstBuilder extends DeltaSqlBaseBaseVisitor[AnyRef] {
Error:(84, 53) not found: type DeltaSqlBaseParser
  protected def parse[T](command: String)(toResult: DeltaSqlBaseParser => T): T = {
Error:(71, 13) value visit is not a member of io.delta.sql.parser.DeltaSqlAstBuilder
    builder.visit(parser.singleStatement()) match {
Error:(85, 21) not found: type DeltaSqlBaseLexer
    val lexer = new DeltaSqlBaseLexer(
Error:(91, 22) not found: type DeltaSqlBaseParser
    val parser = new DeltaSqlBaseParser(tokenStream)
```

screen_shot 2022-03-04 at 18 02 16

## Solution

`Project Settings` > `modules` > `delta-core` > `source folders` > remove any of the `target` folders. should only be `src/main/scala`

Credits to Ah Lag for bringing this to my attention.

### PR DESCRIPTION
**Pull request by scottsand-db**
## Description
- Resolves #988
- Adds an IntelliJ Setup section to README.md which describes how to setup delta-lake in IntelliJ and how to fix a common build error

## How was this patch tested?
Re-cloned Delta Lake and followed the steps myself.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,6 +61,36 @@ To execute tests, run
 
 Refer to [SBT docs](https://www.scala-sbt.org/1.x/docs/Command-Line-Reference.html) for more commands.
 
+# IntelliJ Setup
+
+IntelliJ is the recommended IDE to use when developing Delta Lake. To import Delta Lake as a new project:
+1. Clone Delta Lake into, for example, `~/delta`.
+2. In IntelliJ, select `File` > `New Project` > `Project from Existing Sources...` and select `~/delta`.
+3. Under `Import project from external model` select `sbt`. Click `Next`.
+4. Under `Project JDK` specify a valid Java `1.8` JDK and opt to use SBT shell for `project relaod` and `builds`.
```
Comment (allisonport-db) [inline README.md:70]:
```suggestion
4. Under `Project JDK` specify a valid Java `1.8` JDK and opt to use SBT shell for `project reload` and `builds`.
```

File: README.md [hunk 2]
```diff
@@ -61,6 +61,36 @@ To execute tests, run
...
+After waiting for IntelliJ to index, verify your setup by running a test suite in IntelliJ.
+1. Search for and open `DeltaLogSuite`
+2. Next to the class declaration, right click on the two green arrows and select `Run 'DeltaLogSuite'` 
+
+## Troubleshooting
+
+If you see errors of the form
+
+```
+object DeltaSqlBaseParser is not a member of package io.delta.sql.parser
```
Comment (allisonport-db) [inline README.md:84]: Want to include the "error" part of the message? Just looks a little weird as an error like this.

**Review by allisonport-db — COMMENTED** [review]
Do you think we should reorganize the README structure a little bit? 

Possibly a section like...

**Developers**
   - Building
   - IntelliJ Setup

Or should this go under **Contributing**?

(I would at least move this section to below **Transaction Protocol**.)

What do you think?

**Comment by scottsand-db** [comment]
@allisonport-db - yes we should do something like that. We should look at some other open-source projects for inspiration.

Once we agree on a new format we can make a starter issue?

**Review by allisonport-db — APPROVED** [review]
Can we at least move the `IntelliJ Setup` section below `Transaction Protocol` for now? I feel like since we don't have a table of contents it kind of buries it. Other than that, LGTM.

**Review by allisonport-db — APPROVED** [review]
LGTM

**Pull request closed without merging**